api/object_storage.go

```diff
@@ -0,0 +1,73 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+)
+
+func (a *API) uploadImageWithFormHandler(w http.ResponseWriter, r *http.Request) {
+	// check if the user is authenticated
+	// get the user from the request context
+	user, ok := userFromContext(r.Context())
+	if !ok {
+		ErrUnauthorized.Write(w)
+		return
+	}
+
+	// 32 MB is the default used by FormFile() function
+	if err := r.ParseMultipartForm(32 << 20); err != nil {
+		ErrGenericInternalServerError.With("could not parse form").Write(w)
+		return
+	}
+
+	// Get a reference to the fileHeaders.
+	// They are accessible only after ParseMultipartForm is called
+	files := r.MultipartForm.File["file"]
+	var returnURLs []string
+	for _, fileHeader := range files {
+		// Open the file
+		file, err := fileHeader.Open()
+		if err != nil {
+			ErrGenericInternalServerError.Withf("cannot open file %s", err.Error()).Write(w)
+			break
+		}
+		defer func() {
+			if err := file.Close(); err != nil {
+				ErrGenericInternalServerError.Withf("cannot close file %s", err.Error()).Write(w)
+				return
+			}
+		}()
+		// upload the file using the object storage client
+		// and get the URL of the uploaded file
+		url, err := a.objectStorage.Put(file, fileHeader.Size, user.Email)
+		if err != nil {
+			ErrGenericInternalServerError.Withf("cannot upload file %s", err.Error()).Write(w)
+			break
+		}
+		returnURLs = append(returnURLs, url)
+	}
+	httpWriteJSON(w, map[string][]string{"urls": returnURLs})
+}
+
+func (a *API) downloadImageInlineHandler(w http.ResponseWriter, r *http.Request) {
+	objectID := chi.URLParam(r, "objectName")
```

Comment:
I think that `"objectName"` could be a constant declared in the current file.

Reply:
Hm, not sure I get you. This is not a technique used for the rest of the URI parameters. WHy should we apply it here?

Reply:
I think that is better to have all this strings in a constants file in the same package, no only if they are used multiple times, but also for maintainability.